Internship article Jannick (#77)
alasco-tech/alasco-tech.github.io: _posts/2022-06-20-internship-experience.md

---
author: DeinAlptraum
date: 2022-06-20
layout: post
title: The Alasco internship experience
subtitle: My experience interning for 3 months at Alasco
description: My experience interning for 3 months at Alasco
thumb: internship.jpg
---

### Intro

### My Background

First of all, a little introduction: my name is Jannick, I studied computer science and just finished my Bachelor's thesis at RWTH Aachen in Germany about three months ago. As my course of study did not include any compulsory internships and also little practical exercise otherwise, I was itching to get some real working experience before I move on to do my Master's.

> Looking for internship opportunities in software development, I eventually ended up at Alasco. I work in one of the company's engineering teams on backend side, which means coding in Python and the Django web framework.

Reviewer: I worked as a backend engineer on within one of the engineering teams, which meant coding in Python and the Django web framework.

I have previous experience with Python, using it in university for a bit and contributing to two small projects in my free time. I'm also familiar with Git of course, as well as the command line since I'm a (mostly) Linux user. Django was foregin territory to me though, and I also had no experience working on larger projects in larger teams. As such, a couple challenges lay ahead of me when I began at the start of April.

### Starting Out

> When I was just starting out, everything was quite overwhelming. There's were so many new faces and names to remember, my calendar was full of meetings, using a Mac for the first time takes some getting used to, I had to adapt to a certain workflow for development as well as get to know and understand the code. It was also my first experience working full time, and while that means I have much more free time than back in uni, it is also a much stricter schedule. Frankly, eight hours of work is quite tiring when you're not used to it.

Reviewer: ```suggestion
When I was just starting out, everything was quite overwhelming. There were so many new faces and names to remember as well as my calendar packed full of meetings. Using a Mac for the first time takes some getting used to, I had to adapt to a certain workflow for development as well as get to know and understand the code. Furthermore it was also my first experience working full time, and while that means I have much more free time than back in uni, it is also a much stricter schedule. Frankly, eight hours of work is quite tiring when you're not used to it.
```

> However, most of these were just a queston of time and getting through it. As a newbie, there's little pressure for me to deliver right from the start and I was given the time I need to get comfortable with everything. When I need help with something, I can just ask any of my colleagues and they're happy to help me out. The only person who's bothered when I ask questions, is me.

Reviewer: ```suggestion
However, most of these were just a question of time and getting through it. As a newbie, there's little pressure for me to deliver right from the start and I was given the time I needed to get comfortable with everything. When I need help with something, I can just ask any of my colleagues and they're happy to help me out.
```

### How It's Going

A bit more than two months in and everything is rather smooth sailing at this point. Of course I'm still somewhat slower and need to ask others for help more often than the more experienced developers. But I'm a full member of the team. I take can take over work ranging from small tasks to complete feature stories, seeing them through from tech planning to deployment. I fix bugs reported by customers, clean and refacor ugly old code into shiny new code. I support the frontend engineers on our team when they encounter backend-troubles. I take part in client interviews as a representative of the engineering side, and also take over moderation for some of our team meetings.

There's so many opportunities for me to contribute, and it makes me happy to be able to do my part. As a team, we decide on our overarching goals every six weeks, and I help us reach these goals. I don't feel like "just an intern", but a valued and important part of my team.

> ### Leaning, Learning, Learning

Reviewer: "Leaning" or "Learning"

People help each other out at Alasco, whether you need some advice at work when you're struggling with the code, or need additonal hands to carry all that furniture on your next move. Should you ever suddenly drop out due to disease or similar, your colleagues will cover for you without a second thought. If you need time to recover, everyone wants you to take the time it needs. Work is important, but people come first.

Until the beginning of June, there was no obligation to be in the office. But the people I meet and talk to in the office made it a place where I want to go every day. And every time I step through the door, I'm full of energy with a smile on my face, because I'm happy to see my colleagues. While I can't draw comparisons to other companies, I couldn't have imagined nor asked for anything better.

---

My time at Alasco is coming to an end. Soon, my internship here will be over and I'm setting my sights on my Master. Of course, I could have started right after finishing my Bachelor, but I'm glad I took the time and also got the opportunity for this internship.  

> I believe that over the past months I've improved significantly as a developer. Regardless of what I decide to do for my Master's, I'm sure I'll make good use of this experience, be it for private projects, open source projects or maybe another job. Aside from that, I've also met some amazing and inspiring people, as well as made new friends. And finally, Alasco has shown me what an enjoyable workplace culture should look like

Author: Done 👍
